rust/pact_matching/src/json.rs

```diff
@@ -975,9 +975,8 @@ mod tests {
     let matcher = MatchingRule::ContentType("text/plain".to_string());
     expect!(Value::String("plain text".into()).matches_with(&Value::String("plain text".into()), &matcher, false)).to(be_ok());
     expect!(Value::String("plain text".into()).matches_with(&Value::String("different text".into()), &matcher, false)).to(be_ok());
-    expect!(Value::String("plain text".into()).matches_with(&json!(100), &matcher, false)).to(be_ok());
-    expect!(Value::String("plain text".into()).matches_with(&json!(100.01), &matcher, false)).to(be_ok());
-    #[cfg(not(windows))]
+    expect!(Value::String("plain text".into()).matches_with(&json!(100), &matcher, false)).to(be_err());
+    expect!(Value::String("plain text".into()).matches_with(&json!(100.01), &matcher, false)).to(be_err());
```

Comment:
this is a change in behaviour. was this correct before? I think we should get an error, based on the condition of byte checking in this repo when content type is text/plain but the actual detected content differs.

Reply:
This is the interesting thing about context types. JSON is a text format, so matching `text/plain` with a JSON document should be OK (same goes for other text formats). I.e. If I say I need to get plain text, and you give me JSON, HTML or a Gerhkin file, it is plain text.

Reply:
Apache Tika (which Pact-JVM uses) solves this by creating a hierarchy for the content types. So the parent of `application/json` is `text/plain`.

Reply:
My logic was incorrect, and not the same as the original impl for this case, so it was a bug well caught

Reply:
>Apache Tika (which Pact-JVM uses) solves this by creating a hierarchy for the content types. So the parent of application/json is text/plain.

I laugh now, but I wonder at which point we end up needing to port the logic to Rust, even if just to get consistent behaviour between our two cores

Reply:
I have thought about that many times, but it is not a trivial endeavor.

Reply:
hmm lets cross that bridge if and when we need to come to it!